prisma-fmt/src/references.rs

```diff
@@ -0,0 +1,277 @@
+        .db
+        .walk_models()
+        .chain(ctx.db.walk_views())
+        .find(|model| model.id.1 == model_id && model.file_id() == file_id)
+    else {
+        warn!("Could not find model");
+        return empty_identifiers();
+    };
+
+    let identifier = if let Some(field) = model.scalar_fields().find(|field| field.name() == name) {
+        field.ast_field().identifier()
+    } else {
+        warn!("Could not find field with name: `{}`", name);
+        return empty_identifiers();
+    };
+
+    vec![identifier]
+}
+
+fn find_where_used_for_native_type<'ast>(ctx: &ReferencesContext<'ast>, name: &str) -> Vec<&'ast Identifier> {
+    info!("Get references for native types");
+
+    fn find_native_type_locations<'ast>(
+        name: &str,
+        fields: impl Iterator<Item = (FieldId, &'ast Field)>,
+    ) -> Vec<&'ast Identifier> {
+        fields
+            .filter_map(|field| {
+                field
+                    .1
+                    .attributes
+                    .iter()
+                    .find(|attr| extract_ds_from_native_type(attr.name()) == name)
+                    .map(|attr| attr.identifier())
+            })
+            .collect()
+    }
+
+    ctx.db
+        .walk_tops()
+        .flat_map(|top| match top.ast_top() {
+            Top::CompositeType(composite_type) => find_native_type_locations(name, composite_type.iter_fields()),
+            Top::Model(model) => find_native_type_locations(name, model.iter_fields()),
+
+            Top::Enum(_) | Top::Source(_) | Top::Generator(_) => empty_identifiers(),
+        })
+        .collect()
+}
+
+fn find_where_used_as_type<'ast>(ctx: &'ast ReferencesContext<'_>, name: &str) -> Vec<&'ast Identifier> {
+    info!("Get references for top");
+
+    fn get_relevent_identifiers<'a, 'b>(
+        fields: impl Iterator<Item = (FieldId, &'a Field)>,
+        name: &str,
+    ) -> Vec<&'a Identifier> {
+        fields
+            .filter_map(|(_id, field)| match &field.field_type {
+                FieldType::Supported(id) => {
+                    if id.name == name {
+                        Some(id)
+                    } else {
+                        None
+                    }
+                }
+                FieldType::Unsupported(_, _) => None,
+            })
+            .collect()
+    }
+
+    ctx.db
+        .walk_tops()
+        .flat_map(|top| match top.ast_top() {
+            Top::Model(model) => {
+                info!("Get references in model");
+
+                let fields = model.iter_fields();
+
+                get_relevent_identifiers(fields, name)
+            }
+            Top::CompositeType(composite_type) => {
+                info!("Get references in composite type");
+
+                let fields = composite_type.iter_fields();
+                get_relevent_identifiers(fields, name)
+            }
+            Top::Enum(_) | Top::Source(_) | Top::Generator(_) => empty_identifiers(),
+        })
+        .collect()
```

Comment:
This could use `ctx.db.find_top` instead which avoids iterating through all tops